---
title: "pathway_annotation"
output: html_document
date: "2025-10-15"
---

## Loading of the libraries
```{r setup, include=FALSE}
library(clusterProfiler)
library(readxl)
library(dplyr)
library(tidyr)
library(stringr)
```

## AIM: Convert KEGG pathways id into description of pathways
- pathway identifiers: combination of 2-4 letter prefix code and 5 digit number
- the prefix could have different meaning:
map: manually drawn reference pathway identifiers
ko: reference pathway highlighting KOs

We will take the name that has the prefix ko, since it is easier to search 
the correspondent pathway name

##### T0R1 #####
```{r cars}
# Set working directory
setwd(".../6_Functional_annotation/T0R1/")
getwd()
```
Starting reading the tables, of the gene expression and pathway annotation
```{r pressure, echo=FALSE}
setwd(".../6_Functional_annotation/T0R1/")
kma_table <- read_excel("1_cd_hit_new/kma_modif.xlsx") # the kma output is modified in order
egg_nog_table <- read_excel("egg_nog_annot.xlsx")
```

Take from the eggnog table the name of the query gene
```{r pressure, echo=FALSE}
col_keep_egg <- c("#query", "KEGG_ko", "KEGG_Pathway")
# add the information of the ko names as a 2nd column
sub_eggnog_table <- egg_nog_table[, colnames(egg_nog_table) %in% col_keep_egg]
head(sub_eggnog_table)
```
We want to decide which columns of name to keep, we will see how many NA values there are in the two columns
```{r pressure, echo=FALSE}
library(dplyr)

sub_eggnog_table %>%
  mutate(KEGG_Pathway = as.character(KEGG_Pathway)) %>%
  summarise(count_missing = sum(KEGG_Pathway == "-" | is.na(KEGG_Pathway)))
```
The initial table has a dimension of 56635 and after filtering for na values and "-", the table has 14312

Look at the other column
```{r pressure, echo=FALSE}
sub_eggnog_table %>%
  mutate(KEGG_ko = as.character(KEGG_ko)) %>%
  summarise(count_missing = sum(KEGG_ko == "-" | is.na(KEGG_ko)))
```
There are much more elements missing in the KEGG_Pathway column
```{r pressure, echo=FALSE}
sub_eggnog_table$KEGG_ko <- NULL
```

EGGNOG TABLE:
For each gene i have the correspondent KEGG pathway

Assigning column names to kma table and eggnog table
Important to call with the same name the column Gene in order to do the merge later
```{r pressure, echo=FALSE}
colnames(kma_table) <- c("Genes", "Depth")
colnames(sub_eggnog_table) <- c("Genes", "KEGG_Pathway")
head(kma_table)
```
KMA TABLE:
for each gene i have the correspondent Depth

Modify the gene names, in order to take away the last underscore and the final number after the last underscore
```{r pressure, echo=FALSE}
norm_names_kma <- sub("(_[^_]+)_.*", "\\1", kma_table$Genes)
norm_names_eggnog <- sub("(_[^_]+)_.*", "\\1", sub_eggnog_table$Genes)
head(norm_names_kma)
head(norm_names_eggnog)
```

Reassignment of the gene names
```{r pressure, echo=FALSE}
kma_table$Genes <- norm_names_kma
sub_eggnog_table$Genes <-  norm_names_eggnog
```

Concatenate the pathways present in the same gene and put them in the same row (EGGNOG TABLE)
This is useful because the next step is to unify the informations of the KMA and EGGNOG table, based on the gene names
```{r pressure, echo=FALSE}
# Group by gene and concatenate description
collapsed_eggnog_t0r1 <- sub_eggnog_table %>%
  group_by(Genes) %>% 
  summarise(Description = paste(unique(KEGG_Pathway), collapse = "; "), .groups = "drop")

collapsed_eggnog_new_t0r1 <- collapsed_eggnog_t0r1[-c(1,2,3),]

collapsed_kma_t0r1 <- kma_table %>%
  group_by(Genes) %>%
  summarise(Depth = sum(Depth)) %>%
  ungroup()

head(collapsed_eggnog_new_t0r1) 
head(collapsed_kma_t0r1)
```
Now we take together the two tables based on the gene names
```{r pressure, echo=FALSE}
finaltab_t0r1 <- left_join(collapsed_kma_t0r1, collapsed_eggnog_t0r1, by = "Genes")
dim(collapsed_kma_t0r1)
dim(collapsed_eggnog_t0r1)
dim(finaltab_t0r1)
head(finaltab_t0r1)
```

Take away the entries that in the Description column have na values or "-" sign
```{r pressure, echo=FALSE}
# delete all the rows that have NA values in the column "Description"
finaltab_t0r1_cleaned <- finaltab_t0r1 %>%
  filter(!is.na(Description) & Description != "-")
dim(finaltab_t0r1_cleaned)
head(finaltab_t0r1_cleaned)
```

Finaltab_t0r1_cleaned is the table that has all the informations
```{r pressure, echo=FALSE}
# split in the column Description vector of strings into different rows and delete the id that starts with map
df_cleaned <- finaltab_t0r1_cleaned %>%
  # Split the values column into a list
  separate_rows(Description, sep = "[,;]") %>%
  # Remove values that start with "map"
  filter(!str_starts(Description, "map"))
head(df_cleaned)
```

```{r pressure, echo=FALSE}
df_cleaned_new <- df_cleaned %>%
  filter(Description != " -" & Description != "-")

# trim blank spaces before and after the KEGG ids
df_cleaned_new$Description <- str_trim(df_cleaned_new$Description)
unique(df_cleaned_new$Description)
head(df_cleaned_new)
```

```{r pressure, echo=FALSE}
# now we want to convert this list of KEGG ids into pathways description
# unsing clusterProfiler package

ko.pathways <- download_KEGG(species="ko") # download the repository where there are KEGG terms, ids and names
term2name <- ko.pathways$KEGGPATHID2NAME
colnames(term2name) <- c("Description", "Pathway")
head(term2name)
```
Creation of the final table with the Description and the TotalDepth
```{r pressure, echo=FALSE}
# merge the table df_cleaned_new$Description with term2name table
merged_R1 <- merge(df_cleaned_new, term2name, by="Description")
dim(df_cleaned_new)

# group by pathways, and sum all depths
merged_R1_grouped <- merged_R1 %>%
  group_by(Pathway) %>%
  summarise(TotalDepth = sum(Depth), .groups = "drop")

head(merged_R1_grouped)
```
The dimension of the table is 78548 pathways and for each one three columns

N.B there are too many genes unassigned to pathways
you have to find why, maybe it i better to take another column for example KEGG ko...


### SWT0R2 ###
```{r pressure, echo=FALSE}
# Set working directory
setwd(".../6_Functional_annotation/T0R2")
getwd()

# Read the two tables
kma_table <- read_excel("1_cd_hit_new/kma_mod.xlsx") # the kma output is modified in order
egg_nog_table <- read_excel("1_cd_hit_new/eggnog_annot.xlsx")
```

```{r pressure, echo=FALSE}
# Take from the eggnog table only the name of the contigs and the functional annotation
col_keep_egg <- c("#query", "KEGG_Pathway")
sub_eggnog_table <- egg_nog_table[, colnames(egg_nog_table) %in% col_keep_egg]

# Unify the two tables
colnames(kma_table) <- c("Genes", "Depth")
colnames(sub_eggnog_table) <- c("Genes", "KEGG_Pathway")

norm_names_kma <- sub("(_[^_]+)_.*", "\\1", kma_table$Genes)
norm_names_eggnog <- sub("(_[^_]+)_.*", "\\1", sub_eggnog_table$Genes)

kma_table$Genes <- norm_names_kma
sub_eggnog_table$Genes <-  norm_names_eggnog

# Group by gene and concatenate description
collapsed_eggnog_t0r2 <- sub_eggnog_table %>%
  group_by(Genes) %>%
  summarise(Description = paste(unique(KEGG_Pathway), collapse = "; "), .groups = "drop")

collapsed_eggnog_new_t0r2 <- collapsed_eggnog_t0r2[-c(1,2,3),]

collapsed_kma_t0r2 <- kma_table %>%
  group_by(Genes) %>%
  summarise(Depth = sum(Depth)) %>%
  ungroup()

# now we can concatenate the information of egg-nog and kma for the sample at time T0R1
finaltab_t0r2 <- left_join(collapsed_kma_t0r2, collapsed_eggnog_t0r2, by = "Genes")
dim(collapsed_kma_t0r2)
dim(collapsed_eggnog_t0r2)
dim(finaltab_t0r2)

# delete all the rows that have NA values in the column "Description"
finaltab_t0r2_cleaned <- finaltab_t0r2 %>%
  filter(!is.na(Description) & Description != "-")

# split a vector of strings into different rows and delete the id that starts with map
df_cleaned <- finaltab_t0r2_cleaned %>%
  # Split the values column into a list
  separate_rows(Description, sep = "[,;]") %>%
  # Remove values that start with "map"
  filter(!str_starts(Description, "map"))

df_cleaned_new <- df_cleaned %>%
  filter(Description != " -" & Description != "-")

# trim blank spaces before and after the KEGG ids
df_cleaned_new$Description <- str_trim(df_cleaned_new$Description)
unique(df_cleaned_new$Description)

# now we want to convert this list of kegg ids into pathways description
# unsing clusterProfiler package

ko.pathways <- download_KEGG(species="ko") # download the repository where there are KEGG terms, ids and names
term2name <- ko.pathways$KEGGPATHID2NAME
colnames(term2name) <- c("Description", "Pathway")

# merge the table df_cleaned_new$Description with term2name table
merged_R2 <- merge(df_cleaned_new, term2name, by="Description")
dim(df_cleaned_new)
dim(merged_R2)

merged_R2_grouped <- merged_R2 %>%
  group_by(Pathway) %>%
  summarise(TotalDepth = sum(Depth), .groups = "drop") 
```

# SWT0R3
```{r pressure, echo=FALSE}
# SWT0R3
# Set working directory
setwd(".../6_Functional_annotation/T0R3")
getwd()

library(readxl)
# Read the two tables
kma_table <- read_excel("1_cd_hit_new/kma_depth.xlsx") # the kma output is modified in order
egg_nog_table <- read_excel("1_cd_hit_new/eggnog_annot.xlsx")

# Take from the eggnog table only the name of the contigs and the functional annotation
col_keep_egg <- c("#query", "KEGG_Pathway")
sub_eggnog_table <- egg_nog_table[, colnames(egg_nog_table) %in% col_keep_egg]

# Unify the two tables
colnames(kma_table) <- c("Genes", "Depth")
colnames(sub_eggnog_table) <- c("Genes", "KEGG_Pathway")

norm_names_kma <- sub("(_[^_]+)_.*", "\\1", kma_table$Genes)
norm_names_eggnog <- sub("(_[^_]+)_.*", "\\1", sub_eggnog_table$Genes)

kma_table$Genes <- norm_names_kma
sub_eggnog_table$Genes <-  norm_names_eggnog

# Group by gene and concatenate description
collapsed_eggnog_t0r3 <- sub_eggnog_table %>%
  group_by(Genes) %>%
  summarise(Description = paste(unique(KEGG_Pathway), collapse = "; "), .groups = "drop")

collapsed_eggnog_new_t0r3 <- collapsed_eggnog_t0r3[-c(1,2,3),]

library(dplyr)
collapsed_kma_t0r3 <- kma_table %>%
  group_by(Genes) %>%
  summarise(Depth = sum(Depth)) %>%
  ungroup()

# now we can concatenate the information of egg-nog and kma for the sample at time T0R1
finaltab_t0r3 <- left_join(collapsed_kma_t0r3, collapsed_eggnog_t0r3, by = "Genes")
dim(collapsed_kma_t0r3)
dim(collapsed_eggnog_t0r3)
dim(finaltab_t0r3)

# delete all the rows that have NA values in the column "Description"
finaltab_t0r3_cleaned <- finaltab_t0r3 %>%
  filter(!is.na(Description) & Description != "-")

library(tidyr)
library(stringr)
# split a vector of strings into different rows and delete the id that starts with map
df_cleaned <- finaltab_t0r3_cleaned %>%
  # Split the values column into a list
  separate_rows(Description, sep = "[,;]") %>%
  # Remove values that start with "map"
  filter(!str_starts(Description, "map"))

df_cleaned_new <- df_cleaned %>%
  filter(Description != " -" & Description != "-")

# trim blank spaces before and after the KEGG ids
df_cleaned_new$Description <- str_trim(df_cleaned_new$Description)
unique(df_cleaned_new$Description)

# now we want to convert this list of kegg ids into pathways description
# unsing clusterProfiler package

ko.pathways <- download_KEGG(species="ko") # download the repository where there are KEGG terms, ids and names
term2name <- ko.pathways$KEGGPATHID2NAME
colnames(term2name) <- c("Description", "Pathway")

# merge the table df_cleaned_new$Description with term2name table
merged_R3 <- merge(df_cleaned_new, term2name, by="Description")
dim(df_cleaned_new)
dim(merged_R3)

merged_R3_grouped <- merged_R3 %>%
  group_by(Pathway) %>%
  summarise(TotalDepth = sum(Depth), .groups = "drop") 

# merge the table with the other two tables previously created
mergedT3 <- full_join(merged_R1_grouped, merged_R2_grouped, by="Pathway")
colnames(mergedT3) <- c("Pathways", "R1", "R2")
colnames(merged_R3_grouped) <- c("Pathways", "R3")
mergedT3_new <- full_join(mergedT3, merged_R3_grouped, by="Pathways")
colnames(mergedT3_new) <- c("Pathways", "T0R1", "T0R2", "T0R3")
```

#### SWT1R1 #####
```{r pressure, echo=FALSE}
# Set working directory
setwd(".../6_Functional_annotation/T1R1")
getwd()

library(readxl)
# Read the two tables
kma_table <- read_excel("1_cd_hit_new/kma_annot.xlsx") # the kma output is modified in order
egg_nog_table <- read_excel("1_cd_hit_new/eggnog_annot.xlsx")

# Take from the eggnog table only the name of the contigs and the functional annotation
col_keep_egg <- c("#query", "KEGG_Pathway")
sub_eggnog_table <- egg_nog_table[, colnames(egg_nog_table) %in% col_keep_egg]

# Unify the two tables
colnames(kma_table) <- c("Genes", "Depth")
colnames(sub_eggnog_table) <- c("Genes", "KEGG_Pathway")

#norm_names_kma <- sub("(_[^_]+)_.*", "\\1", kma_table$Genes)
norm_names_eggnog <- sub("(_[^_]+)_.*", "\\1", sub_eggnog_table$Genes)

#kma_table$Genes <- norm_names_kma
sub_eggnog_table$Genes <-  norm_names_eggnog

# Group by gene and concatenate description
collapsed_eggnog_t1r1 <- sub_eggnog_table %>%
  group_by(Genes) %>%
  summarise(Description = paste(unique(KEGG_Pathway), collapse = "; "), .groups = "drop")

collapsed_eggnog_new_t1r1 <- collapsed_eggnog_t1r1[-c(1,2,3),]

library(dplyr)
collapsed_kma_t1r1 <- kma_table %>%
  group_by(Genes) %>%
  summarise(Depth = sum(Depth)) %>%
  ungroup()

# now we can concatenate the information of egg-nog and kma for the sample at time T0R1
finaltab_t1r1 <- left_join(collapsed_kma_t1r1, collapsed_eggnog_t1r1, by = "Genes")
dim(collapsed_kma_t1r1)
dim(collapsed_eggnog_t1r1)
dim(finaltab_t1r1)

# delete all the rows that have NA values in the column "Description"
finaltab_t1r1_cleaned <- finaltab_t1r1 %>%
  filter(!is.na(Description) & Description != "-")

library(tidyr)
library(stringr)
# split a vector of strings into different rows and delete the id that starts with map
df_cleaned <- finaltab_t1r1_cleaned %>%
  # Split the values column into a list
  separate_rows(Description, sep = "[,;]") %>%
  # Remove values that start with "map"
  filter(!str_starts(Description, "map"))

df_cleaned_new <- df_cleaned %>%
  filter(Description != " -" & Description != "-")

# trim blank spaces before and after the KEGG ids
df_cleaned_new$Description <- str_trim(df_cleaned_new$Description)
unique(df_cleaned_new$Description)

# now we want to convert this list of kegg ids into pathways description
# unsing clusterProfiler package

ko.pathways <- download_KEGG(species="ko") # download the repository where there are KEGG terms, ids and names
term2name <- ko.pathways$KEGGPATHID2NAME
colnames(term2name) <- c("Description", "Pathway")

# merge the table df_cleaned_new$Description with term2name table
merged_T1_R1 <- merge(df_cleaned_new, term2name, by="Description")
dim(df_cleaned_new)
dim(merged_T1_R1)

merged_T1_R1_grouped <- merged_T1_R1 %>%
  group_by(Pathway) %>%
  summarise(TotalDepth = sum(Depth), .groups = "drop") 

colnames(mergedT3_new) <- c("Pathway", "T0R1", "T0R2", "T0R3")

merged_t1_r1 <- full_join(merged_T1_R1_grouped, mergedT3_new, by="Pathway")
colnames(merged_t1_r1) <- c("Pathways", "T1R1", "T0R1", "T0R2", "T0R3")
head(merged_t1_r1)
```

##### SWT1R2 ####
```{r pressure, echo=FALSE}
# Set working directory
setwd("/Users/micheladellalma/Library/CloudStorage/OneDrive-UniversitaÌ€degliStudidiMilano/michela_0/Savorgnano/6_Functional_annotation/SWT1R2")
getwd()

library(readxl)
# Read the two tables
kma_table <- read_excel("1_cd_hit_new/kma_out.xlsx") # the kma output is modified in order
egg_nog_table <- read_excel("1_cd_hit_new/eggnog_annot.xlsx")

# Take from the eggnog table only the name of the contigs and the functional annotation
col_keep_egg <- c("#query", "KEGG_Pathway")
sub_eggnog_table <- egg_nog_table[, colnames(egg_nog_table) %in% col_keep_egg]

# Unify the two tables
colnames(kma_table) <- c("Genes", "Depth")
colnames(sub_eggnog_table) <- c("Genes", "KEGG_Pathway")

#norm_names_kma <- sub("(_[^_]+)_.*", "\\1", kma_table$Genes)
norm_names_eggnog <- sub("(_[^_]+)_.*", "\\1", sub_eggnog_table$Genes)

#kma_table$Genes <- norm_names_kma
sub_eggnog_table$Genes <-  norm_names_eggnog

# Group by gene and concatenate description
collapsed_eggnog_t1r2 <- sub_eggnog_table %>%
  group_by(Genes) %>%
  summarise(Description = paste(unique(KEGG_Pathway), collapse = "; "), .groups = "drop")

collapsed_eggnog_new_t1r2 <- collapsed_eggnog_t1r2[-c(1,2,3),]

library(dplyr)
collapsed_kma_t1r2 <- kma_table %>%
  group_by(Genes) %>%
  summarise(Depth = sum(Depth)) %>%
  ungroup()

# now we can concatenate the information of egg-nog and kma for the sample at time T0R1
finaltab_t1r2 <- left_join(collapsed_kma_t1r2, collapsed_eggnog_t1r2, by = "Genes")
dim(collapsed_kma_t1r2)
dim(collapsed_eggnog_t1r2)
dim(finaltab_t1r2)

# delete all the rows that have NA values in the column "Description"
finaltab_t1r2_cleaned <- finaltab_t1r2 %>%
  filter(!is.na(Description) & Description != "-")

library(tidyr)
library(stringr)
# split a vector of strings into different rows and delete the id that starts with map
df_cleaned <- finaltab_t1r2_cleaned %>%
  # Split the values column into a list
  separate_rows(Description, sep = "[,;]") %>%
  # Remove values that start with "map"
  filter(!str_starts(Description, "map"))

df_cleaned_new <- df_cleaned %>%
  filter(Description != " -" & Description != "-")

# trim blank spaces before and after the KEGG ids
df_cleaned_new$Description <- str_trim(df_cleaned_new$Description)
unique(df_cleaned_new$Description)

# now we want to convert this list of kegg ids into pathways description
# unsing clusterProfiler package

ko.pathways <- download_KEGG(species="ko") # download the repository where there are KEGG terms, ids and names
term2name <- ko.pathways$KEGGPATHID2NAME
colnames(term2name) <- c("Description", "Pathway")

# merge the table df_cleaned_new$Description with term2name table
merged_T1_R2 <- merge(df_cleaned_new, term2name, by="Description")
dim(df_cleaned_new)
dim(merged_T1_R2)

merged_T1_R2_grouped <- merged_T1_R2 %>%
  group_by(Pathway) %>%
  summarise(TotalDepth = sum(Depth), .groups = "drop") 

colnames(merged_T1_R2_grouped) <- c("Pathways", "TotalDepth")

merged_t1_r2 <- full_join(merged_T1_R2_grouped, merged_t1_r1, by="Pathways")
colnames(merged_t1_r2)[2] <- c("T1R2")
head(merged_t1_r2)
```

##### SWT1R3 ######
```{r pressure, echo=FALSE}
#### R3 ####
# Set working directory
setwd(".../6_Functional_annotation/T1R3")
getwd()

library(readxl)
# Read the two tables
kma_table <- read_excel("1_cd_hit_new/kma_annot.xlsx") # the kma output is modified in order
egg_nog_table <- read_excel("1_cd_hit_new/eggnog_annot.xlsx")

# Take from the eggnog table only the name of the contigs and the functional annotation
col_keep_egg <- c("#query", "KEGG_Pathway")
sub_eggnog_table <- egg_nog_table[, colnames(egg_nog_table) %in% col_keep_egg]

# Unify the two tables
colnames(kma_table) <- c("Genes", "Depth")
colnames(sub_eggnog_table) <- c("Genes", "KEGG_Pathway")

#norm_names_kma <- sub("(_[^_]+)_.*", "\\1", kma_table$Genes)
norm_names_eggnog <- sub("(_[^_]+)_.*", "\\1", sub_eggnog_table$Genes)

#kma_table$Genes <- norm_names_kma
sub_eggnog_table$Genes <-  norm_names_eggnog

# Group by gene and concatenate description
collapsed_eggnog_t1r3 <- sub_eggnog_table %>%
  group_by(Genes) %>%
  summarise(Description = paste(unique(KEGG_Pathway), collapse = "; "), .groups = "drop")

collapsed_eggnog_new_t1r3 <- collapsed_eggnog_t1r3[-c(1,2,3),]

library(dplyr)
collapsed_kma_t1r3 <- kma_table %>%
  group_by(Genes) %>%
  summarise(Depth = sum(Depth)) %>%
  ungroup()

# now we can concatenate the information of egg-nog and kma for the sample at time T0R1
finaltab_t1r3 <- left_join(collapsed_kma_t1r3, collapsed_eggnog_t1r3, by = "Genes")
dim(collapsed_kma_t1r3)
dim(collapsed_eggnog_t1r3)
dim(finaltab_t1r3)

# delete all the rows that have NA values in the column "Description"
finaltab_t1r3_cleaned <- finaltab_t1r3 %>%
  filter(!is.na(Description) & Description != "-")

library(tidyr)
library(stringr)
# split a vector of strings into different rows and delete the id that starts with map
df_cleaned <- finaltab_t1r3_cleaned %>%
  # Split the values column into a list
  separate_rows(Description, sep = "[,;]") %>%
  # Remove values that start with "map"
  filter(!str_starts(Description, "map"))

df_cleaned_new <- df_cleaned %>%
  filter(Description != " -" & Description != "-")

# trim blank spaces before and after the KEGG ids
df_cleaned_new$Description <- str_trim(df_cleaned_new$Description)
unique(df_cleaned_new$Description)

# now we want to convert this list of kegg ids into pathways description
# unsing clusterProfiler package

ko.pathways <- download_KEGG(species="ko") # download the repository where there are KEGG terms, ids and names
term2name <- ko.pathways$KEGGPATHID2NAME
colnames(term2name) <- c("Description", "Pathway")

# merge the table df_cleaned_new$Description with term2name table
merged_R3 <- merge(df_cleaned_new, term2name, by="Description")
dim(df_cleaned_new)
dim(merged_R3)

merged_R3_grouped <- merged_R3 %>%
  group_by(Pathway) %>%
  summarise(TotalDepth = sum(Depth), .groups = "drop") 

colnames(merged_R3_grouped) <- c("Pathways", "TotalDepth")
merged_t1_r3 <- full_join(merged_R3_grouped, merged_t1_r2, by="Pathways")
colnames(merged_t1_r3)[2] <- "T1R3"
head(merged_t1_r3)
```

##### SWT2R1 #####

# There is a problem with the eggnog annotations of the replicates at time T2, because kegg pathways are wrong
```{r pressure, echo=FALSE}
# Set working directory
setwd(".../6_Functional_annotation/T2R1")
getwd()

library(readxl)
# Read the two tables
kma_table <- read_excel("1_cd_hit_new/kma.xlsx") # the kma output is modified in order
egg_nog_table <- read_excel("1_cd_hit_new/eggnog_annot.xlsx")

# Take from the eggnog table only the name of the contigs and the functional annotation
col_keep_egg <- c("#query", "KEGG_Pathway")
sub_eggnog_table <- egg_nog_table[, colnames(egg_nog_table) %in% col_keep_egg]

# Unify the two tables
colnames(kma_table) <- c("Genes", "Depth")
colnames(sub_eggnog_table) <- c("Genes", "KEGG_Pathway")

#norm_names_kma <- sub("(_[^_]+)_.*", "\\1", kma_table$Genes)
norm_names_eggnog <- sub("(_[^_]+)_.*", "\\1", sub_eggnog_table$Genes)

#kma_table$Genes <- norm_names_kma
sub_eggnog_table$Genes <-  norm_names_eggnog

# Group by gene and concatenate description
collapsed_eggnog_t2r1 <- sub_eggnog_table %>%
  group_by(Genes) %>%
  summarise(Description = paste(unique(KEGG_Pathway), collapse = "; "), .groups = "drop")

library(dplyr)
collapsed_kma_t2r1 <- kma_table %>%
  group_by(Genes) %>%
  summarise(Depth = sum(Depth)) %>%
  ungroup()

# now we can concatenate the information of egg-nog and kma for the sample at time T0R1
finaltab_t2r1 <- left_join(collapsed_kma_t2r1, collapsed_eggnog_t2r1, by = "Genes")
dim(collapsed_kma_t2r1)
dim(collapsed_eggnog_t2r1)
dim(finaltab_t2r1)

# delete all the rows that have NA values in the column "Description"
finaltab_t2r1_cleaned <- finaltab_t2r1 %>%
  filter(!is.na(Description) & Description != "-")

library(tidyr)
library(stringr)
# split a vector of strings into different rows and delete the id that starts with map
df_cleaned <- finaltab_t2r1_cleaned %>%
  # Split the values column into a list
  separate_rows(Description, sep = "[,;]") %>%
  # Remove values that start with "map"
  filter(!str_starts(Description, "map"))

df_cleaned_new <- df_cleaned %>%
  filter(Description != " -" & Description != "-")

# trim blank spaces before and after the KEGG ids
df_cleaned_new$Description <- str_trim(df_cleaned_new$Description)
unique(df_cleaned_new$Description)

# now we want to convert this list of kegg ids into pathways description
# unsing clusterProfiler package

ko.pathways <- download_KEGG(species="ko") # download the repository where there are KEGG terms, ids and names
term2name <- ko.pathways$KEGGPATHID2NAME
colnames(term2name) <- c("Description", "Pathway")

# merge the table df_cleaned_new$Description with term2name table
merged_R1 <- merge(df_cleaned_new, term2name, by="Description")
dim(df_cleaned_new)
dim(merged_R1)

merged_R1_grouped <- merged_R1 %>%
  group_by(Pathway) %>%
  summarise(TotalDepth = sum(Depth), .groups = "drop") 

colnames(merged_R1_grouped) <- c("Pathways", "TotalDepth")
merged_t2_r1 <- full_join(merged_R1_grouped, merged_t1_r3, by="Pathways")
colnames(merged_t2_r1)[2] <- c("T2R1")
head(merged_t2_r1)
```

```{r pressure, echo=FALSE}
#### R2 ####
setwd(".../6_Functional_annotation/T2R2")
getwd()

library(readxl)
# Read the two tables
kma_table <- read_excel("1_cd_hit_new/kma_t2r2.xlsx") # the kma output is modified in order
egg_nog_table <- read_excel("1_cd_hit_new/eggnog_annotation_t2r2.xlsx")

# Take from the eggnog table only the name of the contigs and the functional annotation
col_keep_egg <- c("#query", "KEGG_Pathway")
sub_eggnog_table <- egg_nog_table[, colnames(egg_nog_table) %in% col_keep_egg]

# Unify the two tables
colnames(kma_table) <- c("Genes", "Depth")
colnames(sub_eggnog_table) <- c("Genes", "KEGG_Pathway")

#norm_names_kma <- sub("(_[^_]+)_.*", "\\1", kma_table$Genes)
norm_names_eggnog <- sub("(_[^_]+)_.*", "\\1", sub_eggnog_table$Genes)

#kma_table$Genes <- norm_names_kma
sub_eggnog_table$Genes <-  norm_names_eggnog

# Group by gene and concatenate description
collapsed_eggnog_t2r2 <- sub_eggnog_table %>%
  group_by(Genes) %>%
  summarise(Description = paste(unique(KEGG_Pathway), collapse = "; "), .groups = "drop")

library(dplyr)
collapsed_kma_t2r2 <- kma_table %>%
  group_by(Genes) %>%
  summarise(Depth = sum(Depth)) %>%
  ungroup()

# now we can concatenate the information of egg-nog and kma for the sample at time T0R1
finaltab_t2r2 <- left_join(collapsed_kma_t2r2, collapsed_eggnog_t2r2, by = "Genes")
dim(collapsed_kma_t2r2)
dim(collapsed_eggnog_t2r2)
dim(finaltab_t2r2)

# delete all the rows that have NA values in the column "Description"
finaltab_t2r2_cleaned <- finaltab_t2r2 %>%
  filter(!is.na(Description) & Description != "-")

library(tidyr)
library(stringr)
# split a vector of strings into different rows and delete the id that starts with map
df_cleaned <- finaltab_t2r2_cleaned %>%
  # Split the values column into a list
  separate_rows(Description, sep = "[,;]") %>%
  # Remove values that start with "map"
  filter(!str_starts(Description, "map"))

df_cleaned_new <- df_cleaned %>%
  filter(Description != " -" & Description != "-")

# trim blank spaces before and after the KEGG ids
df_cleaned_new$Description <- str_trim(df_cleaned_new$Description)
unique(df_cleaned_new$Description)

# now we want to convert this list of kegg ids into pathways description
# unsing clusterProfiler package

ko.pathways <- download_KEGG(species="ko") # download the repository where there are KEGG terms, ids and names
term2name <- ko.pathways$KEGGPATHID2NAME
colnames(term2name) <- c("Description", "Pathway")

# merge the table df_cleaned_new$Description with term2name table
merged_R2 <- merge(df_cleaned_new, term2name, by="Description")
dim(df_cleaned_new)
dim(merged_R3)

merged_R2_grouped <- merged_R2 %>%
  group_by(Pathway) %>%
  summarise(TotalDepth = sum(Depth), .groups = "drop") 

colnames(merged_R2_grouped) <- c("Pathways", "TotalDepth")
merged_t2_r2 <- full_join(merged_R2_grouped, merged_t2_r1, by="Pathways")
colnames(merged_t2_r2)[2] <- c("T2R2")
head(merged_t2_r2)
```

#### R3 ####
```{r pressure, echo=FALSE}
setwd(".../6_Functional_annotation/T2R3")
getwd()

library(readxl)
# Read the two tables
kma_table <- read_excel("1_cd_hit_new/kma_t2r3.xlsx") # the kma output is modified in order
egg_nog_table <- read_excel("1_cd_hit_new/eggnog_t2r3.xlsx")

# Take from the eggnog table only the name of the contigs and the functional annotation
col_keep_egg <- c("#query", "KEGG_Pathway")
sub_eggnog_table <- egg_nog_table[, colnames(egg_nog_table) %in% col_keep_egg]

# Unify the two tables
colnames(kma_table) <- c("Genes", "Depth")
colnames(sub_eggnog_table) <- c("Genes", "KEGG_Pathway")

#norm_names_kma <- sub("(_[^_]+)_.*", "\\1", kma_table$Genes)
norm_names_eggnog <- sub("(_[^_]+)_.*", "\\1", sub_eggnog_table$Genes)

#kma_table$Genes <- norm_names_kma
sub_eggnog_table$Genes <-  norm_names_eggnog

# Group by gene and concatenate description
collapsed_eggnog_t2r3 <- sub_eggnog_table %>%
  group_by(Genes) %>%
  summarise(Description = paste(unique(KEGG_Pathway), collapse = "; "), .groups = "drop")

library(dplyr)
collapsed_kma_t2r3 <- kma_table %>%
  group_by(Genes) %>%
  summarise(Depth = sum(Depth)) %>%
  ungroup()

# now we can concatenate the information of egg-nog and kma for the sample at time T0R1
finaltab_t2r3 <- left_join(collapsed_kma_t2r3, collapsed_eggnog_t2r3, by = "Genes")
dim(collapsed_kma_t2r3)
dim(collapsed_eggnog_t2r3)
dim(finaltab_t2r3)

# delete all the rows that have NA values in the column "Description"
finaltab_t2r3_cleaned <- finaltab_t2r3 %>%
  filter(!is.na(Description) & Description != "-")

library(tidyr)
library(stringr)
# split a vector of strings into different rows and delete the id that starts with map
df_cleaned <- finaltab_t2r3_cleaned %>%
  # Split the values column into a list
  separate_rows(Description, sep = "[,;]") %>%
  # Remove values that start with "map"
  filter(!str_starts(Description, "map"))

df_cleaned_new <- df_cleaned %>%
  filter(Description != " -" & Description != "-")

# trim blank spaces before and after the KEGG ids
df_cleaned_new$Description <- str_trim(df_cleaned_new$Description)
unique(df_cleaned_new$Description)

# now we want to convert this list of kegg ids into pathways description
# unsing clusterProfiler package

ko.pathways <- download_KEGG(species="ko") # download the repository where there are KEGG terms, ids and names
term2name <- ko.pathways$KEGGPATHID2NAME
colnames(term2name) <- c("Description", "Pathway")

# merge the table df_cleaned_new$Description with term2name table
merged_R3 <- merge(df_cleaned_new, term2name, by="Description")
dim(df_cleaned_new)
dim(merged_R3)

merged_R3_grouped <- merged_R3 %>%
  group_by(Pathway) %>%
  summarise(TotalDepth = sum(Depth), .groups = "drop") 

colnames(merged_R3_grouped) <- c("Pathways", "TotalDepth")
merged_t2_r3 <- full_join(merged_R3_grouped, merged_t2_r2, by="Pathways")
colnames(merged_t2_r3)[2] <- c("T2R3")
head(merged_t2_r3)
```

## SWT3R1
```{r pressure, echo=FALSE}
# Set working directory
setwd(".../6_Functional_annotation/T3R1/")
getwd()

library(readxl)
# Read the two tables
kma_table <- read_excel("1_cd_hit_new/kma_t3r1.xlsx") # the kma output is modified in order
egg_nog_table <- read_excel("1_cd_hit_new/eggnog_annot.xlsx")

# Take from the eggnog table only the name of the contigs and the functional annotation
col_keep_egg <- c("#query", "KEGG_Pathway")
sub_eggnog_table <- egg_nog_table[, colnames(egg_nog_table) %in% col_keep_egg]

# Unify the two tables
colnames(kma_table) <- c("Genes", "Depth")
colnames(sub_eggnog_table) <- c("Genes", "KEGG_Pathway")

#norm_names_kma <- sub("(_[^_]+)_.*", "\\1", kma_table$Genes)
norm_names_eggnog <- sub("(_[^_]+)_.*", "\\1", sub_eggnog_table$Genes)

#kma_table$Genes <- norm_names_kma
sub_eggnog_table$Genes <-  norm_names_eggnog

# Group by gene and concatenate description
collapsed_eggnog_t3r1 <- sub_eggnog_table %>%
  group_by(Genes) %>%
  summarise(Description = paste(unique(KEGG_Pathway), collapse = "; "), .groups = "drop")

#collapsed_eggnog_new_t0r1 <- collapsed_eggnog_t0r1[-c(1,2,3),]

library(dplyr)
collapsed_kma_t3r1 <- kma_table %>%
  group_by(Genes) %>%
  summarise(Depth = sum(Depth)) %>%
  ungroup()

# now we can concatenate the information of egg-nog and kma for the sample at time T0R1
finaltab_t3r1 <- left_join(collapsed_kma_t3r1, collapsed_eggnog_t3r1, by = "Genes")
dim(collapsed_kma_t3r1)
dim(collapsed_eggnog_t3r1)
dim(finaltab_t3r1)

# delete all the rows that have NA values in the column "Description"
finaltab_t3r1_cleaned <- finaltab_t3r1 %>%
  filter(!is.na(Description) & Description != "-")

library(tidyr)
library(stringr)
# split a vector of strings into different rows and delete the id that starts with map
df_cleaned <- finaltab_t3r1_cleaned %>%
  # Split the values column into a list
  separate_rows(Description, sep = "[,;]") %>%
  # Remove values that start with "map"
  filter(!str_starts(Description, "map"))

df_cleaned_new <- df_cleaned %>%
  filter(Description != " -" & Description != "-")

# trim blank spaces before and after the KEGG ids
df_cleaned_new$Description <- str_trim(df_cleaned_new$Description)
unique(df_cleaned_new$Description)

# now we want to convert this list of kegg ids into pathways description
# unsing clusterProfiler package

ko.pathways <- download_KEGG(species="ko") # download the repository where there are KEGG terms, ids and names
term2name <- ko.pathways$KEGGPATHID2NAME
colnames(term2name) <- c("Description", "Pathway")

# merge the table df_cleaned_new$Description with term2name table
merged_R1 <- merge(df_cleaned_new, term2name, by="Description")
dim(df_cleaned_new)
dim(merged_R1)

merged_R1_grouped <- merged_R1 %>%
  group_by(Pathway) %>%
  summarise(TotalDepth = sum(Depth), .groups = "drop") 

colnames(merged_R1_grouped) <- c("Pathways", "TotalDepth")
merged_t3r1 <- full_join(merged_R1_grouped, merged_t2_r3, by="Pathways")
colnames(merged_t3r1)[2] <- "T3R1" 
head(merged_t3r1)
```

## SWT3R2
```{r pressure, echo=FALSE}
# Set working directory
setwd(".../6_Functional_annotation/T3R2/")
getwd()

library(readxl)
# Read the two tables
kma_table <- read_excel("1_cd_hit_new/kma_t3r2.xlsx") # the kma output is modified in order
egg_nog_table <- read_excel("1_cd_hit_new/t3r2_eggnog.xlsx")

# Take from the eggnog table only the name of the contigs and the functional annotation
col_keep_egg <- c("#query", "KEGG_Pathway")
sub_eggnog_table <- egg_nog_table[, colnames(egg_nog_table) %in% col_keep_egg]

# Unify the two tables
colnames(kma_table) <- c("Genes", "Depth")
colnames(sub_eggnog_table) <- c("Genes", "KEGG_Pathway")

#norm_names_kma <- sub("(_[^_]+)_.*", "\\1", kma_table$Genes)
norm_names_eggnog <- sub("(_[^_]+)_.*", "\\1", sub_eggnog_table$Genes)

#kma_table$Genes <- norm_names_kma
sub_eggnog_table$Genes <-  norm_names_eggnog

# Group by gene and concatenate description
collapsed_eggnog_t3r2 <- sub_eggnog_table %>%
  group_by(Genes) %>%
  summarise(Description = paste(unique(KEGG_Pathway), collapse = "; "), .groups = "drop")

#collapsed_eggnog_new_t0r1 <- collapsed_eggnog_t0r1[-c(1,2,3),]

library(dplyr)
collapsed_kma_t3r2 <- kma_table %>%
  group_by(Genes) %>%
  summarise(Depth = sum(Depth)) %>%
  ungroup()

# now we can concatenate the information of egg-nog and kma for the sample at time T0R1
finaltab_t3r2 <- left_join(collapsed_kma_t3r2, collapsed_eggnog_t3r2, by = "Genes")
dim(collapsed_kma_t3r2)
dim(collapsed_eggnog_t3r2)
dim(finaltab_t3r2)

# delete all the rows that have NA values in the column "Description"
finaltab_t3r2_cleaned <- finaltab_t3r2 %>%
  filter(!is.na(Description) & Description != "-")

library(tidyr)
library(stringr)
# split a vector of strings into different rows and delete the id that starts with map
df_cleaned <- finaltab_t3r2_cleaned %>%
  # Split the values column into a list
  separate_rows(Description, sep = "[,;]") %>%
  # Remove values that start with "map"
  filter(!str_starts(Description, "map"))

df_cleaned_new <- df_cleaned %>%
  filter(Description != " -" & Description != "-")

# trim blank spaces before and after the KEGG ids
df_cleaned_new$Description <- str_trim(df_cleaned_new$Description)
unique(df_cleaned_new$Description)

# now we want to convert this list of kegg ids into pathways description
# unsing clusterProfiler package

ko.pathways <- download_KEGG(species="ko") # download the repository where there are KEGG terms, ids and names
term2name <- ko.pathways$KEGGPATHID2NAME
colnames(term2name) <- c("Description", "Pathway")

# merge the table df_cleaned_new$Description with term2name table
merged_R2 <- merge(df_cleaned_new, term2name, by="Description")
dim(df_cleaned_new)
dim(merged_R2)

merged_R2_grouped <- merged_R2 %>%
  group_by(Pathway) %>%
  summarise(TotalDepth = sum(Depth), .groups = "drop") 

colnames(merged_R2_grouped) <- c("Pathways", "Depth")

merged_t3r2 <- full_join(merged_R2_grouped, merged_t3r1, by="Pathways")
colnames(merged_t3r2)[2] <- "T3R2"
head(merged_t3r2)
```

## SWT3R3
```{r pressure, echo=FALSE}
# Set working directory
setwd(".../6_Functional_annotation/T3R3/")
getwd()

library(readxl)
# Read the two tables
kma_table <- read_excel("1_cd_hit_new/kma_t3r3.xlsx") # the kma output is modified in order
egg_nog_table <- read_excel("1_cd_hit_new/t3r3_eggnog.xlsx")

# Take from the eggnog table only the name of the contigs and the functional annotation
col_keep_egg <- c("#query", "KEGG_Pathway")
sub_eggnog_table <- egg_nog_table[, colnames(egg_nog_table) %in% col_keep_egg]

# Unify the two tables
colnames(kma_table) <- c("Genes", "Depth")
colnames(sub_eggnog_table) <- c("Genes", "KEGG_Pathway")

#norm_names_kma <- sub("(_[^_]+)_.*", "\\1", kma_table$Genes)
norm_names_eggnog <- sub("(_[^_]+)_.*", "\\1", sub_eggnog_table$Genes)

#kma_table$Genes <- norm_names_kma
sub_eggnog_table$Genes <-  norm_names_eggnog

# Group by gene and concatenate description
collapsed_eggnog_t3r3 <- sub_eggnog_table %>%
  group_by(Genes) %>%
  summarise(Description = paste(unique(KEGG_Pathway), collapse = "; "), .groups = "drop")

collapsed_eggnog_new_t3r3 <- collapsed_eggnog_t3r3[-c(1,2,3),]

library(dplyr)
collapsed_kma_t3r3 <- kma_table %>%
  group_by(Genes) %>%
  summarise(Depth = sum(Depth)) %>%
  ungroup()

# now we can concatenate the information of egg-nog and kma for the sample at time T0R1
finaltab_t3r3 <- left_join(collapsed_kma_t3r3, collapsed_eggnog_t3r3, by = "Genes")
dim(collapsed_kma_t3r3)
dim(collapsed_eggnog_t3r3)
dim(finaltab_t3r3)

# delete all the rows that have NA values in the column "Description"
finaltab_t3r3_cleaned <- finaltab_t3r3 %>%
  filter(!is.na(Description) & Description != "-")

library(tidyr)
library(stringr)
# split a vector of strings into different rows and delete the id that starts with map
df_cleaned <- finaltab_t3r3_cleaned %>%
  # Split the values column into a list
  separate_rows(Description, sep = "[,;]") %>%
  # Remove values that start with "map"
  filter(!str_starts(Description, "map"))

df_cleaned_new <- df_cleaned %>%
  filter(Description != " -" & Description != "-")

# trim blank spaces before and after the KEGG ids
df_cleaned_new$Description <- str_trim(df_cleaned_new$Description)
unique(df_cleaned_new$Description)

# now we want to convert this list of kegg ids into pathways description
# unsing clusterProfiler package

ko.pathways <- download_KEGG(species="ko") # download the repository where there are KEGG terms, ids and names
term2name <- ko.pathways$KEGGPATHID2NAME
colnames(term2name) <- c("Description", "Pathway")

# merge the table df_cleaned_new$Description with term2name table
merged_R3 <- merge(df_cleaned_new, term2name, by="Description")
dim(df_cleaned_new)
dim(merged_R3)

merged_R3_grouped <- merged_R3 %>%
  group_by(Pathway) %>%
  summarise(TotalDepth = sum(Depth), .groups = "drop") 

colnames(merged_R3_grouped) <- c("Pathways", "Depth")
  
merged_t3r3 <- full_join(merged_R3_grouped, merged_t3r2, by="Pathways")
colnames(merged_t3r3)[2] <- c("T3R3")
head(merged_t3r3)
```

##### SWT5 #####
#### R1 ####
```{r pressure, echo=FALSE}
# Set working directory
setwd(".../6_Functional_annotation/T5R1")
getwd()

library(readxl)
# Read the two tables
kma_table <- read_excel("1_cd_hit_new/t5r1_kma.xlsx") # the kma output is modified in order
egg_nog_table <- read_excel("1_cd_hit_new/t5r1_eggnog.xlsx")

# Take from the eggnog table only the name of the contigs and the functional annotation
col_keep_egg <- c("#query", "KEGG_Pathway")
sub_eggnog_table <- egg_nog_table[, colnames(egg_nog_table) %in% col_keep_egg]

# Unify the two tables
colnames(kma_table) <- c("Genes", "Depth")
colnames(sub_eggnog_table) <- c("Genes", "KEGG_Pathway")

#norm_names_kma <- sub("(_[^_]+)_.*", "\\1", kma_table$Genes)
norm_names_eggnog <- sub("(_[^_]+)_.*", "\\1", sub_eggnog_table$Genes)

#kma_table$Genes <- norm_names_kma
sub_eggnog_table$Genes <-  norm_names_eggnog

# Group by gene and concatenate description
collapsed_eggnog_t5r1 <- sub_eggnog_table %>%
  group_by(Genes) %>%
  summarise(Description = paste(unique(KEGG_Pathway), collapse = "; "), .groups = "drop")

collapsed_eggnog_new_t5r1 <- collapsed_eggnog_t5r1[-c(1,2,3),]

library(dplyr)
collapsed_kma_t5r1 <- kma_table %>%
  group_by(Genes) %>%
  summarise(Depth = sum(Depth)) %>%
  ungroup()

# now we can concatenate the information of egg-nog and kma for the sample at time T0R1
finaltab_t5r1 <- left_join(collapsed_kma_t5r1, collapsed_eggnog_t5r1, by = "Genes")
dim(collapsed_kma_t5r1)
dim(collapsed_eggnog_t5r1)
dim(finaltab_t5r1)

# delete all the rows that have NA values in the column "Description"
finaltab_t5r1_cleaned <- finaltab_t5r1 %>%
  filter(!is.na(Description) & Description != "-")

library(tidyr)
library(stringr)
# split a vector of strings into different rows and delete the id that starts with map
df_cleaned <- finaltab_t5r1_cleaned %>%
  # Split the values column into a list
  separate_rows(Description, sep = "[,;]") %>%
  # Remove values that start with "map"
  filter(!str_starts(Description, "map"))

df_cleaned_new <- df_cleaned %>%
  filter(Description != " -" & Description != "-")

# trim blank spaces before and after the KEGG ids
df_cleaned_new$Description <- str_trim(df_cleaned_new$Description)
unique(df_cleaned_new$Description)

# now we want to convert this list of kegg ids into pathways description
# unsing clusterProfiler package

ko.pathways <- download_KEGG(species="ko") # download the repository where there are KEGG terms, ids and names
term2name <- ko.pathways$KEGGPATHID2NAME
colnames(term2name) <- c("Description", "Pathway")

# merge the table df_cleaned_new$Description with term2name table
merged_R1 <- merge(df_cleaned_new, term2name, by="Description")
dim(df_cleaned_new)
dim(merged_R1)

merged_R1_grouped <- merged_R1 %>%
  group_by(Pathway) %>%
  summarise(TotalDepth = sum(Depth), .groups = "drop") 

colnames(merged_R1_grouped) <- c("Pathways", "Depth")

merged_t5r1 <- full_join(merged_R1_grouped, merged_t3r3, by="Pathways")
colnames(merged_t5r1)[2] <- c("T5R1")
head(merged_t5r1)
```

```{r pressure, echo=FALSE}
# Set working directory
setwd(".../6_Functional_annotation/T5R2")
getwd()

library(readxl)
# Read the two tables
kma_table <- read_excel("1_cd_hit_new/kma_t5r2.xlsx") # the kma output is modified in order
egg_nog_table <- read_excel("1_cd_hit_new/t5r2_eggnog.xlsx")

# Take from the eggnog table only the name of the contigs and the functional annotation
col_keep_egg <- c("#query", "KEGG_Pathway")
sub_eggnog_table <- egg_nog_table[, colnames(egg_nog_table) %in% col_keep_egg]

# Unify the two tables
colnames(kma_table) <- c("Genes", "Depth")
colnames(sub_eggnog_table) <- c("Genes", "KEGG_Pathway")

#norm_names_kma <- sub("(_[^_]+)_.*", "\\1", kma_table$Genes)
norm_names_eggnog <- sub("(_[^_]+)_.*", "\\1", sub_eggnog_table$Genes)

#kma_table$Genes <- norm_names_kma
sub_eggnog_table$Genes <-  norm_names_eggnog

# Group by gene and concatenate description
collapsed_eggnog_t5r2 <- sub_eggnog_table %>%
  group_by(Genes) %>%
  summarise(Description = paste(unique(KEGG_Pathway), collapse = "; "), .groups = "drop")

collapsed_eggnog_new_t5r2 <- collapsed_eggnog_t5r2[-c(1,2,3),]

library(dplyr)
collapsed_kma_t5r2 <- kma_table %>%
  group_by(Genes) %>%
  summarise(Depth = sum(Depth)) %>%
  ungroup()

# now we can concatenate the information of egg-nog and kma for the sample at time T0R1
finaltab_t5r2 <- left_join(collapsed_kma_t5r2, collapsed_eggnog_t5r2, by = "Genes")
dim(collapsed_kma_t5r2)
dim(collapsed_eggnog_t5r2)
dim(finaltab_t5r2)

# delete all the rows that have NA values in the column "Description"
finaltab_t5r2_cleaned <- finaltab_t5r2 %>%
  filter(!is.na(Description) & Description != "-")

library(tidyr)
library(stringr)
# split a vector of strings into different rows and delete the id that starts with map
df_cleaned <- finaltab_t5r2_cleaned %>%
  # Split the values column into a list
  separate_rows(Description, sep = "[,;]") %>%
  # Remove values that start with "map"
  filter(!str_starts(Description, "map"))

df_cleaned_new <- df_cleaned %>%
  filter(Description != " -" & Description != "-")

# trim blank spaces before and after the KEGG ids
df_cleaned_new$Description <- str_trim(df_cleaned_new$Description)
unique(df_cleaned_new$Description)

# now we want to convert this list of kegg ids into pathways description
# unsing clusterProfiler package

ko.pathways <- download_KEGG(species="ko") # download the repository where there are KEGG terms, ids and names
term2name <- ko.pathways$KEGGPATHID2NAME
colnames(term2name) <- c("Description", "Pathway")

# merge the table df_cleaned_new$Description with term2name table
merged_R2 <- merge(df_cleaned_new, term2name, by="Description")
dim(df_cleaned_new)
dim(merged_R2)

merged_R2_grouped <- merged_R2 %>%
  group_by(Pathway) %>%
  summarise(TotalDepth = sum(Depth), .groups = "drop") 
colnames(merged_R2_grouped) <- c("Pathways", "Depth")
merged_t5r2 <- full_join(merged_R2_grouped, merged_t5r1, by="Pathways")
colnames(merged_t5r2)[2] <- "T5R2"
head(merged_t5r2)
```
## SWT5R3   
```{r pressure, echo=FALSE}
# Set working directory
setwd(".../6_Functional_annotation/T5R3")
getwd()

library(readxl)
# Read the two tables
kma_table <- read_excel("1_cd_hit_new/kma_depth.xlsx") # the kma output is modified in order
egg_nog_table <- read_excel("1_cd_hit_new/eggnog_annot.xlsx")

# Take from the eggnog table only the name of the contigs and the functional annotation
col_keep_egg <- c("#query", "KEGG_Pathway")
sub_eggnog_table <- egg_nog_table[, colnames(egg_nog_table) %in% col_keep_egg]

# Unify the two tables
colnames(kma_table) <- c("Genes", "Depth")
colnames(sub_eggnog_table) <- c("Genes", "KEGG_Pathway")

#norm_names_kma <- sub("(_[^_]+)_.*", "\\1", kma_table$Genes)
norm_names_eggnog <- sub("(_[^_]+)_.*", "\\1", sub_eggnog_table$Genes)

#kma_table$Genes <- norm_names_kma
sub_eggnog_table$Genes <-  norm_names_eggnog

# Group by gene and concatenate description
collapsed_eggnog_t5r3 <- sub_eggnog_table %>%
  group_by(Genes) %>%
  summarise(Description = paste(unique(KEGG_Pathway), collapse = "; "), .groups = "drop")

collapsed_eggnog_new_t5r3 <- collapsed_eggnog_t5r3[-c(1,2,3),]

library(dplyr)
collapsed_kma_t5r3 <- kma_table %>%
  group_by(Genes) %>%
  summarise(Depth = sum(Depth)) %>%
  ungroup()

# now we can concatenate the information of egg-nog and kma for the sample at time T0R1
finaltab_t5r3 <- left_join(collapsed_kma_t5r3, collapsed_eggnog_t5r3, by = "Genes")
dim(collapsed_kma_t5r3)
dim(collapsed_eggnog_t5r3)
dim(finaltab_t5r3)

# delete all the rows that have NA values in the column "Description"
finaltab_t5r3_cleaned <- finaltab_t5r3 %>%
  filter(!is.na(Description) & Description != "-")

library(tidyr)
library(stringr)
# split a vector of strings into different rows and delete the id that starts with map
df_cleaned <- finaltab_t5r3_cleaned %>%
  # Split the values column into a list
  separate_rows(Description, sep = "[,;]") %>%
  # Remove values that start with "map"
  filter(!str_starts(Description, "map"))

df_cleaned_new <- df_cleaned %>%
  filter(Description != " -" & Description != "-")

# trim blank spaces before and after the KEGG ids
df_cleaned_new$Description <- str_trim(df_cleaned_new$Description)
unique(df_cleaned_new$Description)

# now we want to convert this list of kegg ids into pathways description
# unsing clusterProfiler package

ko.pathways <- download_KEGG(species="ko") # download the repository where there are KEGG terms, ids and names
term2name <- ko.pathways$KEGGPATHID2NAME
colnames(term2name) <- c("Description", "Pathway")

# merge the table df_cleaned_new$Description with term2name table
merged_R3 <- merge(df_cleaned_new, term2name, by="Description")
dim(df_cleaned_new)
dim(merged_R3)

merged_R3_grouped <- merged_R3 %>%
  group_by(Pathway) %>%
  summarise(TotalDepth = sum(Depth), .groups = "drop") 
colnames(merged_R3_grouped) <- c("Pathways", "Depth")
merged_t5r3 <- full_join(merged_R3_grouped, merged_t5r2, by="Pathways")
colnames(merged_t5r3)[2] <- "T5R3"
head(merged_t5r3)
```

## Complete table
```{r pressure, echo=FALSE}
# now we will reorder the columns, and after that we will do:
# - overview of functional pathways detected
# - temporal trends and differential expression of pathways across stages
# - notable pathways linked to fermentation
merged_t5r3_new <- as.data.frame(merged_t5r3[,c("Pathways","T0R1", "T0R2", "T0R3", "T1R1", "T1R2", "T1R3", 
                                                "T2R1", "T2R2", "T2R3", "T3R1", "T3R2", "T3R3", "T5R1", "T5R2", "T5R3")])

# substitute NA values to zeros
merged_t5r3_new[is.na(merged_t5r3_new)] <- 0
rownames(merged_t5r3_new) <- merged_t5r3$Pathways
merged_t5r3_new$Pathways <- NULL
```

```{r pressure, echo=FALSE}
head(merged_t5r3_new)
```
```{r pressure, echo=FALSE}
colnames(merged_t5r3_new)[1] <- "T0R1"
head(merged_t5r3_new)
```

Save the table in an excel file
```{r pressure, echo=FALSE}
library(writexl)
merged_t5r3_new_2 <- cbind(RowName=rownames(merged_t5r3_new), merged_t5r3_new)
path <- ".../8_Corr_Pathways_Metabolites/table_pathways.xlsx"
write_xlsx(merged_t5r3_new_2, path, col_names = TRUE, format_headers = TRUE)
```

## DESeq2
```{r pressure, echo=FALSE}
library(DESeq2)
# set the condition name
condition <- factor(c("T0", "T0", "T0", 
                      "T1", "T1", "T1", 
                      "T2", "T2", "T2", 
                      "T3", "T3", "T3", 
                      "T5", "T5", "T5"))

# transform all the values in the cells in integers
merged_t5r3_new <- data.frame(lapply(merged_t5r3_new, as.integer))
rownames(merged_t5r3_new) <- merged_t5r3$Pathways

dds <- DESeqDataSetFromMatrix(countData = merged_t5r3_new,
                              colData = data.frame(condition),
                              design = ~ condition)

# Run DESeq2 analysis
dds <- DESeq(dds)

# Obtain results for differential expression
res <- results(dds)

# View summary of results
summary(res)

# Filter for significantly different pathways
significant_res <- res[which(res$padj < 0.05 & abs(res$log2FoldChange) > 1), ]

# View the significant pathways
head(significant_res)

significant_pathways <- rownames(significant_res)

# compare T1 vs T0
res_T1_vs_T0 <- results(dds, contrast=c("condition", "T1", "T0"))

# Compare T2 vs T1
res_T2_vs_T1 <- results(dds, contrast = c("condition", "T2", "T1"))

# Compare T3 vs T0
res_T3_vs_T2 <- results(dds, contrast = c("condition", "T3", "T2"))

# Compate T5 vs T3
res_T5_vs_T3 <- results(dds, contrast = c("condition", "T5", "T3"))

#### T0 VS T1 #####
# Filter significant pathways (padj < 0.05 and |log2FoldChange| > 1)
significant_pathways_T1_vs_T0 <- res_T1_vs_T0[which(res_T1_vs_T0$padj < 0.05 & abs(res_T1_vs_T0$log2FoldChange) > 1), ]

# Select significant pathways for heatmap visualization
significant_pathways_t10 <- rownames(significant_pathways_T1_vs_T0)

# we decide to normalize count for visualization, 
# because we see there is variability between replicates and so we want to reduce this variability
# in particular we use the function rlog() that transforms the count data to the log2 scale in a way 
# which minimizes differences between samples for rows with small counts,
# and which normalizes respect to library size.
# The rlog transformation produces a similar variance stabilizing effect.
rld <- rlog(dds, blind = FALSE)  # blind = FALSE keeps group structure
rld_mat <- assay(rld)
```

## Heatmap of significant pathways
# T0 vs T1
```{r pressure, echo=FALSE}
library(pheatmap)
pheatmap(rld_mat[significant_pathways_t10, c("T0R1", "T0R2", "T0R3", "T1R1", "T1R2", "T1R3")], 
         cluster_rows = TRUE, 
         cluster_cols = FALSE, 
         show_rownames = TRUE, 
         show_colnames = TRUE,
         scale = "row",  # Standardize pathways
         color = colorRampPalette(c("blue", "white", "red"))(50),
         main="Pathway difference T0 vs T1")
```

## T1 vs T2
# Filter significant pathways (padj < 0.05 and |log2FoldChange| > 1)
```{r pressure, echo=FALSE}
significant_pathways_T2_vs_T1 <- res_T2_vs_T1[which(res_T2_vs_T1$padj < 0.05 & abs(res_T2_vs_T1$log2FoldChange) > 1), ]

# Select significant pathways for heatmap visualization
significant_pathways_t12 <- rownames(significant_pathways_T2_vs_T1)

# Create a heatmap of significant pathways
library(pheatmap)
pheatmap(rld_mat[significant_pathways_t12, c("T1R1", "T1R2", "T1R3", "T2R1", "T2R2", "T2R3")], 
         cluster_rows = TRUE, 
         cluster_cols = FALSE, 
         show_rownames = TRUE, 
         show_colnames = TRUE,
         scale = "row",  # Standardize pathways
         color = colorRampPalette(c("blue", "white", "red"))(50),
         main="Pathway difference between T1 vs T2")
```

## T2 vs T3
```{r pressure, echo=FALSE}
# no significant pathways for T2 vs T3
# Filter significant pathways (padj < 0.05 and |log2FoldChange| > 1)
significant_pathways_T2_vs_T3 <- res_T3_vs_T2[which(res_T3_vs_T2$padj < 0.05 & abs(res_T3_vs_T2$log2FoldChange) > 1), ]
# no significant pathways between T2 and T3

# Select significant pathways for heatmap visualization
significant_pathways_23 <- rownames(significant_pathways_T2_vs_T3)

# Create a heatmap of significant pathways
library(pheatmap)
pheatmap(rld_mat[significant_pathways, c("T2R1", "T2R2", "T2R3", "T3R1", "T3R2", "T3R3")], 
         cluster_rows = TRUE, 
         cluster_cols = FALSE, 
         show_rownames = TRUE, 
         show_colnames = TRUE,
         scale = "row",  # Standardize pathways
         color = colorRampPalette(c("blue", "white", "red"))(50),
         main="Pathway difference between T2 vs T3")
```

## T3 vs T5
```{r pressure, echo=FALSE}
# Filter significant pathways (padj < 0.05 and |log2FoldChange| > 1)
significant_pathways_T5_vs_T3 <- res_T5_vs_T3[which(res_T5_vs_T3$padj < 0.05 & abs(res_T5_vs_T3$log2FoldChange) > 1), ]

# Select significant pathways for heatmap visualization
significant_pathways_35 <- rownames(significant_pathways_T5_vs_T3)

# Create a heatmap of significant pathways
library(pheatmap)
pheatmap(rld_mat[significant_pathways_35, c("T3R1", "T3R2", "T3R3", "T5R1", "T5R2", "T5R3")], 
         cluster_rows = TRUE, 
         cluster_cols = FALSE, 
         show_rownames = TRUE, 
         show_colnames = TRUE,
         scale = "row",  # Standardize pathways
         color = colorRampPalette(c("blue", "white", "red"))(50),
         main = "Pathway difference between T3 VS T5")

# Take all the pathways that result significant from the differential analysis and
# See the temporal trend over time
signif_path_tot <- c(significant_pathways_t10, significant_pathways_t12, significant_pathways_23, significant_pathways_35)
head(signif_path_tot)

# Subset from the initial dataframe only that specific pathways
sub_merged <- merged_t5r3_new[rownames(merged_t5r3_new) %in% signif_path_tot, ]
dim(sub_merged)
```
```{r pressure, echo=FALSE}
sign_path_all <- as.data.frame(c(significant_pathways_t10, significant_pathways_t12, significant_pathways_23,
significant_pathways_35))
```

```{r pressure, echo=FALSE}
setwd("/Users/micheladellalma/Library/CloudStorage/OneDrive-UniversitaÌ€degliStudidiMilano/michela_0/Savorgnano/8_Corr_Pathways_Metabolites")
write_xlsx(sign_path_all, "all_significant_pathways.xlsx")
```

## Bubble plots
### BUBBLE PLOTS: analysis of potential pathways into different timings of fermentation ##########

```{r pressure, echo=FALSE}
# multiple bubble plots on one line
library(ggplot2)
facet_wrap(~Comparison, scales="free_x")

# before binding by rows all the tables, we add for each table the column "Comparison"
significant_pathways_T1_vs_T0$Comparison <- "T0 vs T1"
significant_pathways_T2_vs_T1$Comparison <- "T1 vs T2"
# significant_pathways_T2_vs_T3$Comparison <- "T2 vs T3"  # we have to delete this comparison since there are not significant different pathways
significant_pathways_T5_vs_T3$Comparison <- "T3 vs T5"

significant_pathways_T1_vs_T0$Pathways <- rownames(significant_pathways_T1_vs_T0)
significant_pathways_T2_vs_T1$Pathways <- rownames(significant_pathways_T2_vs_T1)
significant_pathways_T5_vs_T3$Pathways <- rownames(significant_pathways_T5_vs_T3)

all_comparisons <- rbind(
  significant_pathways_T1_vs_T0,
  significant_pathways_T2_vs_T1,
  significant_pathways_T5_vs_T3
)


all_comparisons$Comparison <- factor(
  all_comparisons$Comparison,
  levels = c("T0 vs T1", "T1 vs T2", "T3 vs T5")
)

all_comparisons$Pathways <- rownames(all_comparisons)

all_comparisons$direction <- ifelse(all_comparisons$log2FoldChange > 0, "latter", "former")

ggplot(all_comparisons, aes(x = baseMean, y = Pathways)) +
  geom_point(aes(size = -log10(pvalue), color = direction), alpha = 0.8) +
  scale_color_manual(values = c("former" = "orange", "latter" = "green"),
                     labels = c("former" = "Enriched in former", "latter" = "Enriched in latter")) +
  scale_size_continuous(
    range = c(2, 9.5),  # Adjust range to make bubbles visibly larger
    name = "-log10(p-value)"  # Optional: more informative legend
  ) +
  facet_wrap(~Comparison, scales = "free_x") +
  theme_minimal() +
  labs(
    title = "Pathway Activity Across Time Points",
    x = "Depth",
    y = "Pathway annotation",
    color = "Direction"
  ) + 
  theme(
    legend.position = "right",
    strip.text = element_text(face = "bold"),
    axis.text.y = element_text(size = 9)
  )
```

## TO vs T1
```{r pressure, echo=FALSE}
# convert the pathways from rownames to a new column 
significant_pathways_T1_vs_T0$Pathways <- rownames(significant_pathways_T1_vs_T0)
head(significant_pathways_T1_vs_T0)

significant_pathways_T1_vs_T0$log_p <- -log10(significant_pathways_T1_vs_T0$pvalue)

significant_pathways_T1_vs_T0$direction <- ifelse(significant_pathways_T1_vs_T0$log2FoldChange > 0, "latter", "former")

significant_pathways_T1_vs_T0$Comparison <- "T0 vs T1"

# Bubble plot
library(ggplot2)
ggplot(significant_pathways_T1_vs_T0, aes(x = baseMean, y = Pathways)) +
  geom_point(aes(size = log_p, color = direction), alpha = 0.8) +
  scale_color_manual(values = c("former" = "orange", "latter" = "green"),
                     labels = c("former" = "Enriched in former", "latter" = "Enriched in latter")) +
  facet_wrap(~Comparison, scales = "free_y") +
  theme_minimal() +
  labs(
    title = "T0 vs T1 Pathway Activity",
    x = "Depth",
    y = "Pathway annotation",
    size = "-log10(p-value)",
    color = "Direction"
  ) + 
  theme(
    legend.position = "right",
    strip.text = element_text(face = "bold"),
    axis.text.y = element_text(size = 9)
  )
```
Among these, consider only the significant pathways interesting for the alcoholic fermentation process and represent them in a histogram 
```{r pressure, echo=FALSE}
# recover the significant pathways that are related to alcoholic fermentation
significant_pathways_T1_vs_T0$Pathways

fermentation_pathways <- c(
  "ABC transporters", # linked to biofilm formation
  "Bacterial secretion system",
  "Biofilm formation - Pseudomonas aeruginosa",
  "Caffeine metabolism",
  "Carotenoid biosynthesis",
  "Cell cycle - Caulobacter",
  "Chlorocyclohexane and chlorobenzene degradation",
  "Fluorobenzoate degradation",
  "Quorum sensing",
  "Toluene degradation",
  "Lipopolysaccharide biosynthesis",
  "beta-Lactam resistance",
  "Dioxin degradation",
  "Nonribosomal peptide structures",
  "Peptidoglycan biosynthesis",
  "Phosphotransferase system (PTS)",
  "Vancomycin resistance",
  "Ethylbenzene degradation"
)

# take for this pathways the values os basemean in time t0 and t1 taking into account also the mean and standard deviation
sub_merged_t0t1 <- merged_t5r3_new[which(rownames(merged_t5r3_new) %in% fermentation_pathways),]
dim(sub_merged_t0t1) # finally i have 6 pathways and 15 samples, for now we take only the ones at time T0 and T1
sub_merged_t0t1 <- sub_merged_t0t1[,c("T0R1", "T0R2", "T0R3", "T1R1","T1R2", "T1R3")]
head(sub_merged_t0t1)
```

```{r pressure, echo=FALSE}
library(dplyr)
library(tibble)

df_summary <- sub_merged_t0t1 %>%
  tibble::rownames_to_column("Pathway") %>%   # ðŸ‘ˆ save rownames into a "Pathway" column
  rowwise() %>%
  mutate(
    T0_mean = mean(c(T0R1, T0R2, T0R3)),
    T1_mean = mean(c(T1R1, T1R2, T1R3)),
    T0_sd   = sd(c(T0R1, T0R2, T0R3)),
    T1_sd   = sd(c(T1R1, T1R2, T1R3))
  ) %>%
  ungroup()

```

```{r pressure, echo=FALSE}
library(dplyr)
library(tidyr)

df_long <- df_summary %>%
  # bring rownames into a proper column
  select(Pathway, T0_mean, T1_mean, T0_sd, T1_sd) %>%
  pivot_longer(
    cols = starts_with("T"),
    names_to = c("Time", ".value"),
    names_pattern = "(T[01])_(.*)"
  )

```

```{r pressure, echo=FALSE}
library(ggplot2)
library(dplyr)
 
# now we log transform the values in the table beacuse otherwise 
# we cannot see the difference in values between time points 
# since the values are very different

# Add log10 transformed mean and approximate SE for error bars
df_summary <- df_long %>%
  mutate(
    log_mean = log10(mean + 1),       # add 1 to avoid log(0)
    # Approximate SE on log scale using delta method:
    log_se = sd / (mean * log(10) + 1e-8)  # adding small number to avoid div by zero
  )

ggplot(df_summary, aes(x = Pathway, y = log_mean, fill = Time)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.8)) +
  geom_errorbar(aes(ymin = log_mean - log_se, ymax = log_mean + log_se),
                width = 0.2,
                position = position_dodge(width = 0.8)) +
  scale_fill_manual(values = c("T0" = "orange", "T1" = "green")) +  # Set colors here
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 11)  # increase size here
  ) +
  labs(
    y = "Log10 Mean Expression (+1)",
    x = "Pathway",
    title = "Pathway expression at T0 vs T1 (log10 scale)",
    fill = "Time"
  )

```

## T1 VS T2
```{r pressure, echo=FALSE}
significant_pathways_T2_vs_T1$Pathways <- rownames(significant_pathways_T2_vs_T1)
head(significant_pathways_T2_vs_T1)

significant_pathways_T2_vs_T1$log_p <- -log10(significant_pathways_T2_vs_T1$pvalue)

significant_pathways_T2_vs_T1$direction <- ifelse(significant_pathways_T2_vs_T1$log2FoldChange > 0, "latter", "former")

significant_pathways_T2_vs_T1$Comparison <- "T1 vs T2"

ggplot(significant_pathways_T2_vs_T1, aes(x = baseMean, y = Pathways)) +
  geom_point(aes(size = log_p, color = direction), alpha = 0.8) +
  scale_color_manual(values = c("former" = "orange", "latter" = "green"),
                     labels = c("former" = "Enriched in former", "latter" = "Enriched in latter")) +
  theme_minimal() +
  facet_wrap(~Comparison, scales = "free_y") +
  labs(
    title = "T1 vs T2 Pathway Activity",
    x = "Depth",
    y = "Pathway annotation",
    size = "-log10(p-value)",
    color = "Direction"
  ) + 
  theme(
    legend.position = "right",
    strip.text = element_text(face = "bold"),
    axis.text.y = element_text(size = 9)
  )
```
Among these, consider only the significant pathways interesting for the alcoholic fermentation process and represent them in a histogram 
```{r pressure, echo=FALSE}
# recover the significant pathways that are related to alcoholic fermentation
significant_pathways_T2_vs_T1$Pathways

fermentation_pathways <- c(
  "Bacterial secretion system",
  "ABC transporters",
  "Polycyclic aromatic hydrocarbon degradation",
  "Naphthalene degradation",
  "Furfural degradation",
  "Ethylbenzene degradation",
  "Dioxin degradation",
  "Cell cycle - Caulobacter",
  "Biofilm formation - Pseudomonas aeruginosa",
  "Bacterial secretion system",
  "ABC transporters"
)

# take for this pathways the values os basemean in time t0 and t1 taking into account also the mean and standard deviation
sub_merged_t1t2 <- merged_t5r3_new[which(rownames(merged_t5r3_new) %in% fermentation_pathways),]
dim(sub_merged_t1t2) # finally i have 6 pathways and 15 samples, for now we take only the ones at time T0 and T1
sub_merged_t1t2 <- sub_merged_t1t2[,c("T1R1", "T1R2", "T1R3", "T2R1","T2R2", "T2R3")]
head(sub_merged_t1t2)
```

```{r pressure, echo=FALSE}
library(dplyr)

df_summary <- sub_merged_t1t2 %>%
  rowwise() %>%
  mutate(
    T1_mean = mean(c(T1R1, T1R2, T1R3)),
    T2_mean = mean(c(T2R1, T2R2, T2R3)),
    T1_sd   = sd(c(T1R1, T1R2, T1R3)),
    T2_sd   = sd(c(T2R1, T2R2, T2R3))
  ) %>%
  ungroup()
df_summary$Pathway <- rownames(sub_merged_t1t2)
```

```{r pressure, echo=FALSE}
library(dplyr)
library(tidyr)

df_long <- df_summary %>%
  # bring rownames into a proper column
  select(Pathway, T1_mean, T2_mean, T1_sd, T2_sd) %>%
  pivot_longer(
    cols = starts_with("T"),
    names_to = c("Time", ".value"),
    names_pattern = "(T[12])_(.*)"
  )

```

```{r pressure, echo=FALSE}
library(ggplot2)
library(dplyr)
 
# now we log transform the values in the table beacuse otherwise 
# we cannot see the difference in values between time points 
# since the values are very different

# Add log10 transformed mean and approximate SE for error bars
df_summary <- df_long %>%
  mutate(
    log_mean = log10(mean + 1),       # add 1 to avoid log(0)
    # Approximate SE on log scale using delta method:
    log_se = sd / (mean * log(10) + 1e-8)  # adding small number to avoid div by zero
  )

ggplot(df_summary, aes(x = Pathway, y = log_mean, fill = Time)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.8)) +
  geom_errorbar(aes(ymin = log_mean - log_se, ymax = log_mean + log_se),
                width = 0.2,
                position = position_dodge(width = 0.8)) +
  scale_fill_manual(values = c("T1" = "orange", "T2" = "green")) +  # Set colors here
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 13)  # increase size here
  ) +
  labs(
    y = "Log10 Mean Expression (+1)",
    x = "Pathway",
    title = "Pathway expression at T1 vs T2 (log10 scale)",
    fill = "Time"
  )

```

##### T3 vs T5 #####
```{r pressure, echo=FALSE}
significant_pathways_T5_vs_T3$Pathways <- rownames(significant_pathways_T5_vs_T3)
head(significant_pathways_T5_vs_T3)

significant_pathways_T5_vs_T3$log_p <- -log10(significant_pathways_T5_vs_T3$pvalue)

significant_pathways_T5_vs_T3$direction <- ifelse(significant_pathways_T5_vs_T3$log2FoldChange > 0, "latter", "former")

significant_pathways_T5_vs_T3$Comparison <- "T3 vs T5"

ggplot(significant_pathways_T5_vs_T3, aes(x = baseMean, y = Pathways)) +
  geom_point(aes(size = log_p, color = direction), alpha = 0.8) +
  scale_color_manual(values = c("former" = "orange", "latter" = "green"),
                     labels = c("former" = "Enriched in former", "latter" = "Enriched in latter")) +
  facet_wrap(~Comparison, scales = "free_y") +
  theme_minimal() +
  labs(
    title = "T3 vs T5 Pathway Activity",
    x = "Depth",
    y = "Pathway annotation",
    size = "-log10(p-value)",
    color = "Direction"
  ) + 
  theme(
    legend.position = "right",
    strip.text = element_text(face = "bold"),
    axis.text.y = element_text(size = 9)
  )
```
Among these, consider only the significant pathways interesting for the alcoholic fermentation process and represent them in a histogram 
```{r pressure, echo=FALSE}
# recover the significant pathways that are related to alcoholic fermentation
significant_pathways_T5_vs_T3$Pathways

fermentation_pathways <- c(
  "Biofilm formation - Pseudomonas aeruginosa",
  "Caffeine metabolism",
  "Cell cycle - Caulobacter",
  "Oxidative phosphorylation",
  "Two-component system",
  "Penicillin and cephalosporin biosynthesis",
  "Ethylbenzene degradation",
  "Furfural degradation"
)

# take for this pathways the values os basemean in time t0 and t1 taking into account also the mean and standard deviation
sub_merged_t3t5 <- merged_t5r3_new[which(rownames(merged_t5r3_new) %in% fermentation_pathways),]
dim(sub_merged_t3t5) # finally i have 6 pathways and 15 samples, for now we take only the ones at time T0 and T1
sub_merged_t3t5 <- sub_merged_t3t5[,c("T3R1", "T3R2", "T3R3", "T5R1","T5R2", "T5R3")]
head(sub_merged_t3t5)
```

```{r pressure, echo=FALSE}
library(dplyr)

df_summary <- sub_merged_t3t5 %>%
  rowwise() %>%
  mutate(
    T3_mean = mean(c(T3R1, T3R2, T3R3)),
    T5_mean = mean(c(T5R1, T5R2, T5R3)),
    T3_sd   = sd(c(T3R1, T3R2, T3R3)),
    T5_sd   = sd(c(T5R1, T5R2, T5R3))
  ) %>%
  ungroup()
df_summary$Pathway <- rownames(sub_merged_t3t5)
```

```{r pressure, echo=FALSE}
library(dplyr)
library(tidyr)

df_long <- df_summary %>%
  # bring rownames into a proper column
  select(Pathway, T3_mean, T5_mean, T3_sd, T5_sd) %>%
  pivot_longer(
    cols = starts_with("T"),
    names_to = c("Time", ".value"),
    names_pattern = "(T[35])_(.*)"
  )

```

```{r pressure, echo=FALSE}
library(ggplot2)
library(dplyr)
 
# now we log transform the values in the table beacuse otherwise 
# we cannot see the difference in values between time points 
# since the values are very different

# Add log10 transformed mean and approximate SE for error bars
df_summary <- df_long %>%
  mutate(
    log_mean = log10(mean + 1),       # add 1 to avoid log(0)
    # Approximate SE on log scale using delta method:
    log_se = sd / (mean * log(10) + 1e-8)  # adding small number to avoid div by zero
  )

ggplot(df_summary, aes(x = Pathway, y = log_mean, fill = Time)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.8)) +
  geom_errorbar(aes(ymin = log_mean - log_se, ymax = log_mean + log_se),
                width = 0.2,
                position = position_dodge(width = 0.8)) +
  scale_fill_manual(values = c("T3" = "orange", "T5" = "green")) +  # Set colors here
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 13)  # increase size here
  ) +
  labs(
    y = "Log10 Mean Expression (+1)",
    x = "Pathway",
    title = "Pathway expression at T3 vs T5 (log10 scale)",
    fill = "Time"
  )
```

Creation of a file with all the pathways that are significant and interesting (those that we plotted in the histograms)
```{r pressure, echo=FALSE}
pathways <- c(rownames(sub_merged_t0t1), rownames(sub_merged_t1t2), rownames(sub_merged_t3t5))
length(pathways)
unique_pathways <- unique(pathways)
length(unique_pathways)
```

Save in a file the unique pathways
```{r pressure, echo=FALSE}
setwd(".../8_Corr_Pathways_Metabolites")
getwd()
df_unique_path <- as.data.frame(unique_pathways)
write_xlsx(df_unique_path, "unique_pathways.xlsx")

setwd(".../9_Corr_Pathways_MO/")
write_xlsx(df_unique_path, "unique_pathways.xlsx")
```